### PR TITLE
JP-985: Update calwebb_spec3 for non-TSO NIRISS SOSS

### PR DESCRIPTION
**Pull request by hbushouse**
Updated the ``calwebb_spec3`` pipeline module to handle NIRISS SOSS (NIS_SOSS) the same it already handles NIRCam and NIRISS WFSS exposures, which is to simply run ``extract_1d`` and ``combine_1d`` to produce a combined x1d product from all input members (no resampling, no outlier detection, ...). NIRISS SOSS data should only ever go through the ``calwebb_spec3`` pipeline when they are NOT obtained in TSO mode (TSO mode SOSS data go through the ``calwebb_tso3`` pipeline).

Also added entries to the default ``calwebb_spec3.cfg`` file for the ``combine_1d`` step.

### CODE REVIEW

**Comment by hbushouse** [comment]
Fixes #4017

**Comment by hbushouse** [comment]
Documentation updates fix https://jira.stsci.edu/browse/JP-1089 and #4183

File: docs/jwst/pipeline/calwebb_spec3.rst [hunk 1]
```diff
@@ -80,8 +95,8 @@ Source-based calibrated data
 
 For NIRSpec fixed-slit, NIRSpec MOS, and NIRCam and NIRISS WFSS, which have a defined
 set of slits or sources, the data from the input calibrated exposures is reorganized
-by :ref:`exp_to_source <exp_to_source>` so that all of the instances of data for a
-particular source/slit are contained in a
+by the :ref:`exp_to_source <exp_to_source>` step so that all of the instances of data
```
Comment (philhodge) [inline docs/jwst/pipeline/calwebb_spec3.rst]: Just a question:  Why is "_step" included within the angle brackets for other step names but not for this one?

Comment (hbushouse) [reply docs/jwst/pipeline/calwebb_spec3.rst]: Just randomness of how the cross-reference labels were created/spelled in the corresponding rst files for the steps.

**Pull request merged**